src/ngraph_mark_for_clustering.cc

```diff
@@ -69,17 +74,33 @@ static inline void SetStaticInputs(Node* n, std::vector<int32> inputs) {
 // the right.
 static ConfirmationFunction SimpleConfirmationFunction(
     const std::vector<int32>& static_input_indices = {}) {
-  auto cf = [static_input_indices](Node* n, bool* result) {
+#if (TF_VERSION_GEQ_1_11)
+  auto cf = [static_input_indices, deadness](Node* n, bool* result) {
     // Adjust negative input indices.
```

Comment:
Move the deadness to inside this lambda. This is the safest thing to do for shared pointers. See this for details: https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

Reply:
@avijit-nervana DeadnessAnalysis is run on the whole graph and the result stored in this 'deadness' ptr. Then we were using it in SimpleConfirmationFunction() to check deadness for each node <line 396>. MarkForClustering(Graph*) is the only function with access to graph. 

So, moving deadness inside this lambda wasn't an option as deadness needs to be used in MarkForClustering(Graph*) function. 

To avoid global initialization, as pointed out,
1. Moved the deadness_analyzer initialization to MarkForClustering(Graph*) function
2. Added a separate function for each check(Ngraph Placement, Deadness Analysis, Input type constraint, confirmation checks) we make on the node, before marking for clustering (Some refactoring).  
 
Tried to create a design that could be easily extended if we want to add more checks in future or update what each check does.

Happy to try something else, if we don't want this refactoring now.